lunatrace/bsl/ingest-worker/pkg/vulnerability/affected/ingester.go

```diff
@@ -0,0 +1,79 @@
+package affected
+
+import (
+	"context"
+
+	"github.com/Khan/genqlient/graphql"
+	"github.com/google/uuid"
+	"github.com/rs/zerolog/log"
+	"go.uber.org/fx"
+
+	"github.com/lunasec-io/lunasec/lunatrace/bsl/ingest-worker/pkg/metadata"
+	"github.com/lunasec-io/lunasec/lunatrace/gogen/gql"
+	"github.com/lunasec-io/lunasec/lunatrace/gogen/gql/types"
+)
+
+type Ingester interface {
+	Ingest(ctx context.Context, vulnerabilityId string) error
+}
+
+type ingesterParams struct {
+	fx.In
+
+	GQLClient  graphql.Client
+	Ingester   metadata.PackageIngester
+	Replicator metadata.APIReplicator
+}
+
+type ingester struct {
+	ingesterParams
+}
+
+func (i ingester) Ingest(ctx context.Context, vulnerabilityId string) error {
+	vulnUUID, err := uuid.Parse(vulnerabilityId)
+	if err != nil {
+		log.Error().
+			Err(err).
+			Msg("unable to parse vulnerability id")
+		return err
+	}
+
+	resp, err := gql.GetVulnerabilityMetadata(ctx, i.GQLClient, vulnUUID)
+	if err != nil {
+		log.Error().
+			Err(err).
+			Msg("unable to get vulnerability metadata")
+		return err
+	}
+	for _, affected := range resp.Vulnerability_by_pk.Affected {
+		if affected.Package.Package_manager == types.NPM {
+			_, err = i.Ingester.Ingest(ctx, affected.Package.Name)
```

Comment:
I assume this is calling into the npm ingester, but I am getting a bit lost here because everything is just called "ingest" and when you have all these packages exporting identically named stuff, it gets confusing. I think maybe sometime in the future we could rename each ingester to "NpmIngester" and "VulnIngester" and so on

Reply:
Otherwise this is really dope though.